src/lib/tvgPaint.h

```diff
@@ -148,14 +147,31 @@ namespace tvg
 
             auto newFlag = static_cast<RenderUpdateFlag>(pFlag | flag);
             flag = RenderUpdateFlag::None;
+            void *edata = nullptr;
+            void *compEData = nullptr;
+
+            if (this->comp && compMethod == CompMethod::ClipPath) {
+                compEData = this->comp->pImpl->update(renderer, pTransform, compList, static_cast<RenderUpdateFlag>(pFlag | flag));
```

Comment:
1. auto edata = this->comp-> ...

2. It looks pTransform must be multiplied with current paint's transform 
see 163 line how it does.

Reply:
compEngineData that was pushed should be popped at line 171.

Reply:
Transform of composite target should not be affected by current paint's transform